flow/cmd/api.go

```diff
@@ -168,13 +174,72 @@ func APIMain(ctx context.Context, args *APIServerParams) error {
 
 	slog.Info(fmt.Sprintf("Starting API gateway on port %d", args.GatewayPort))
 	go func() {
-		if err := gateway.ListenAndServe(); err != nil {
+		if err := gateway.ListenAndServe(); err != nil && err != http.ErrServerClosed {
 			log.Fatalf("failed to serve http: %v", err)
 		}
 	}()
 
-	<-ctx.Done()
+	// somewhat unrelated, need a process which isn't replicated
+	// migrate peer protos to current encryption key
+	go func() {
```

Comment:
can we do this before flow worker establishes a connection w/ temporal etc? Can we add info logs around:
1. If we are skipping or current key id is empty.
2. previous and current key id for a peer.
3. Successfully encrypted.

Reply:
discussed aside to leave async operation